django_tables2/views.py

```diff
@@ -15,18 +16,22 @@ class TableMixinBase(object):
     '''
 
     context_table_name = 'table'
+    table_class = None
     table_pagination = None
 
     def get_table_class(self):
         '''
         Return the class to use for the table.
         '''
-        if self.table_class:
-            return self.table_class
-        klass = type(self).__name__
-        raise ImproperlyConfigured(
-            'A table class was not specified. Define {}.table_class'.format(klass)
-        )
+        if self.table_class is None:
+            if self.model is None:
+                klass = type(self).__name__
+                raise ImproperlyConfigured(
+                    "You must either specify {0}.table_class or"
+                    "{0}.table_class".format(klass)
+                )
+            self.table_class = tables.table_factory(self.model)
+        return self.table_class
```

Comment:
Can you undo the nesting here:
```
if self.table_class:
   return self.table_class
if self.model:
   return table_factory(self.model)

raise ImproperlyConfigured(...)
```

I think saving to self.table_class is not needed.

Also: the error message has `table_class` twice, one must be `model`.